File: Delphi/LocalSQLScript/LocalSqlForm.pas
unit LocalSqlForm;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, dbcClasses, dbcSQL_Exec, dbcLocalSQLExec, dbcDBEngine,
  dbcConnection_BDE, DB, DBTables, StdCtrls, Buttons, ComCtrls;

type
  TForm1 = class(TForm)
    Database: TDatabase;
    DBCConnectionBDE: TDBCConnectionBDE;
    LocalSQLExec: TLocalSQLExec;
    EditFileName: TEdit;
    BitBtn1: TBitBtn;
    OpenDialog: TOpenDialog;
    MemoScript: TRichEdit;
    BitBtn2: TBitBtn;
    MemoLog: TRichEdit;
    ProgressBar: TProgressBar;
    procedure BitBtn1Click(Sender: TObject);
    procedure ShowErrorMessage(Sender: TObject;
      ErrText: String);
    procedure LocalSQLExecLogNextLine(Sender: TObject; LogText: String);
    procedure BitBtn2Click(Sender: TObject);
    procedure LocalSQLExecProgressUpdate(Sender: TObject; Value,
      MaxValue: Integer);
  private
    { Private declarations }
  public
    { Public declarations }
    procedure AddToLog(aText: String);
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.BitBtn1Click(Sender: TObject);
begin
  if OpenDialog.Execute then begin
    EditFileName.Text := OpenDialog.FileName;
    if FileExists(OpenDialog.FileName) then begin
      LocalSQLExec.LoadFromFile(OpenDialog.FileName);
      MemoScript.Lines.LoadFromFile(OpenDialog.FileName);
      MemoLog.Lines.Clear;
      AddToLog('Script loaded: '+ IntToStr(LocalSQLExec.StatementsCount)+' statements.');
    end
    else AddToLog('File not found: '+OpenDialog.FileName);
  end;
end;

procedure TForm1.AddToLog(aText: String);
begin
  MemoLog.Lines.Add(aText);
  MemoLog.Perform(EM_SCROLLCARET, 0, 0);
end;


procedure TForm1.ShowErrorMessage(Sender: TObject;
  ErrText: String);
begin
  AddToLog('Error: '+ ErrText);
end;

procedure TForm1.LocalSQLExecLogNextLine(Sender: TObject; LogText: String);
begin
  AddToLog(LogText);
end;

procedure TForm1.BitBtn2Click(Sender: TObject);
begin
  DBCConnectionBDE.Connected := True;
  LocalSQLExec.ExecuteScript();
  DBCConnectionBDE.Connected := False;
end;

procedure TForm1.LocalSQLExecProgressUpdate(Sender: TObject; Value,
  MaxValue: Integer);
begin
  ProgressBar.Max := MaxValue;
  ProgressBar.Position := Value;
end;

end.

File: CBuilder/LocalSQLScript/LocalSqlForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "LocalSqlForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "dbcClasses"
#pragma link "dbcConnection_BDE"
#pragma link "dbcDBEngine"
#pragma link "dbcLocalSQLExec"
#pragma link "dbcSQL_Exec"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::BitBtn1Click(TObject *Sender)
{
  if (OpenDialog->Execute())
  {
    EditFileName->Text = OpenDialog->FileName;
    if (FileExists(OpenDialog->FileName))
    {
      LocalSQLExec->LoadFromFile(OpenDialog->FileName);
      MemoScript->Lines->LoadFromFile(OpenDialog->FileName);
      MemoLog->Lines->Clear();
      AddToLog("Script loaded: " + IntToStr(LocalSQLExec->StatementsCount) + " statements.");
    } else
    {
      AddToLog("File not found: " + OpenDialog->FileName);
    }
  }
}
//---------------------------------------------------------------------------

void __fastcall TForm1::ShowErrorMessage(TObject *Sender, UnicodeString ErrText)
{
  AddToLog("Error: " + ErrText);
}
//---------------------------------------------------------------------------

void __fastcall TForm1::LocalSQLExecLogNextLine(TObject *Sender, UnicodeString LogText)
{
  AddToLog(LogText);
}
//---------------------------------------------------------------------------

void __fastcall TForm1::BitBtn2Click(TObject *Sender)
{
  DBCConnectionBDE->Connected = true;
  LocalSQLExec->ExecuteScript();
  DBCConnectionBDE->Connected = false;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::LocalSQLExecProgressUpdate(TObject *Sender, int Value, int MaxValue)
{
  ProgressBar->Max = MaxValue;
  ProgressBar->Position = Value;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::AddToLog(UnicodeString aText)
{
  MemoLog->Lines->Add(aText);
  MemoLog->Perform(EM_SCROLLCARET, 0, 0);
}
//---------------------------------------------------------------------------


